### PR TITLE
Add Bitbucket Cloud provider

### PR DESCRIPTION
**Pull request by LoremFooBar**
I know no one asked for it, but seemed pretty straightforward to add support for Bitbucket Cloud so I did.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/Nerdbank.GitVersioning?pullRequest=789) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/Nerdbank.GitVersioning?pullRequest=789) before we can accept your contribution.<br/><br/>:x: LoremFooBar [sign now](https://cla.dotnetfoundation.org/dotnet/Nerdbank.GitVersioning?pullRequest=789)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/Nerdbank.GitVersioning?pullRequest=789) it.</sub>

**Comment by AArnott** [comment]
The failure in AzP looks unrelated to your change. I've queued a fresh build of the `main` branch to confirm whether your PR is somehow causing this to fail.

File: src/NerdBank.GitVersioning/CloudBuildServices/BitbucketCloud.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/// <summary>
+/// Cloud build handling for Bitbucket Cloud.
+/// </summary>
+/// <remarks>
+/// The Bitbucket-specific properties referenced here are <see href="https://support.atlassian.com/bitbucket-cloud/docs/variables-and-secrets/">documented here</see>.
+/// </remarks>
+public class BitbucketCloud : ICloudBuild
+{
+    /// <inheritdoc />
+    public bool IsApplicable => !string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("BITBUCKET_REPO_SLUG"));
```
[the remark this answers is not in the document]
Comment (LoremFooBar) [reply src/NerdBank.GitVersioning/CloudBuildServices/BitbucketCloud.cs:15]: I see your point. If we really want to be on the safe side, I could test for multiple environment variables that are not commonly used:
BITBUCKET_PIPELINE_UUID
BITBUCKET_STEP_UUID
BITBUCKET_STEP_TRIGGERER_UUID

The chance of anyone defining all of these variables outside of an actual Bitbucket Pipelines environment is incredibly low.

[the remark this answers is not in the document]
Comment (AArnott) [reply src/NerdBank.GitVersioning/CloudBuildServices/BitbucketCloud.cs:15]: Thank you for calling out this risk, @KalleOlaviNiemitalo

**Comment by LoremFooBar** [comment]
> The failure in AzP looks unrelated to your change. I've queued a fresh build of the `main` branch to confirm whether your PR is somehow causing this to fail.

I see the pipeline for `main` passed. Looking at the logs I am not sure what exactly failed, is there anything I can try locally? If you think this is a transient error, I will eventually push a fix for @KalleOlaviNiemitalo's comment so it will trigger another run that will hopefully pass.

**Comment by AArnott** [comment]
I do think this is a transient error. It appears my request to rebuild your PR got passed the prior failing point. I see you marked this PR as ready to merge, but are you planning to address the env var risk that's in the active thread?

**Comment by LoremFooBar** [comment]
@AArnott sorry, missed a failed push. But seems we have the same pipeline issue again...

**Pull request merged**